scanner/laravel.go

```diff
@@ -104,3 +114,54 @@ func extractPhpVersion() (string, error) {
 
 	return "", fmt.Errorf("could not find php version")
 }
+
+func extractConnections(path string) (DatabaseKind, bool, bool) {
+	/*
+		Determine the default db
+		Determine whether redis connection is desired
+			returns ( db, redis, skipDB )
+	*/
+
+	// Get File Content
+	file, err := os.Open(path)
+	if err != nil {
+		return 0, false, true
+	}
+	defer file.Close() //skipcq: GO-S2307
+
+	// Set up Regex to match
+	// -not commented out, with DB_CONNECTION
+	dbReg := regexp.MustCompile("^ *DB_CONNECTION *= *[a-zA-Z]+")
+	// -not commented out with redis keyword
+	redisReg := regexp.MustCompile("^[^#]*redis")
+
+	// Default Return Variables
+	var db DatabaseKind = 0
+	redis := false
+	skipDb := true
+
+	// Check each line for
+	// match on redis or db regex
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		text := scanner.Text()
+
+		if redisReg.MatchString(text) {
+			redis = true
+			skipDb = false
+		} else if db == 0 && dbReg.MatchString(text) {
+			if strings.Contains(text, "mysql") {
+				db = DatabaseKindMySQL
+				skipDb = false
+			} else if strings.Contains(text, "pgsql") {
+				db = DatabaseKindPostgres
+				skipDb = false
+			} else if strings.Contains(text, "sqlite") {
+				db = DatabaseKindSqlite
+				skipDb = false
```

Comment:
text contains is very loose matching, a better approach would URI parsing. See https://gobyexample.com/url-parsing

Reply:
Laravel's config isn't a connection string that'll work with parseURL, - in this case `DB_CONNECTION` is one of `mysql, `pgsql`, `sqlite`, `sqlsrv` and in rarer cases, something custom. It's not a full DB connection string tho.

Reply:
<img width="374" alt="image" src="https://github.com/superfly/flyctl/assets/467411/3877646d-6946-46bb-896a-6f49fc026e1f">

Reply:
Team thanks!! I added the DATABSE_URL case in the regex to check connections from!